lib/active_model/serializer/field.rb

```diff
@@ -0,0 +1,56 @@
+module ActiveModel
+  class Serializer
+    # Holds all the meta-data about a field (i.e. attribute or association) as it was
+    # specified in the ActiveModel::Serializer class.
+    # Notice that the field block is evaluated in the context of the serializer.
+    Field = Struct.new(:name, :options, :block) do
+      # Compute the actual value of a field for a given serializer instance.
+      # @param [Serializer] The serializer instance for which the value is computed.
+      # @return [Object] value
+      #
+      # @api private
+      #
+      def value(serializer)
+        if block
+          serializer.instance_eval(&block)
+        else
+          serializer.read_attribute_for_serialization(name)
+        end
+      end
+
+      # Decide whether the field should be serialized by the given serializer instance.
+      # @param [Serializer] The serializer instance
+      # @return [Bool]
+      #
+      # @api private
+      #
+      def included?(serializer)
```

Comment:
nitpick, not a blocker:

I think `excluded?` would be better for two reasons

1. included sounds like include
1. the usage e.g. `next unless reflection.included?(self)` and  implementation, which defaults to true, both point to the method's purpose to exclude. 

I'd be ok with just making `def excluded?(serializer); !included?(serializer); end` and using changing the usage code to e.g. `next if attr.excluded?(self)`. I think the `included?` implementation is easier to reason about, but `excluded?` is the better name to use...